README.rst

```diff
@@ -17,18 +17,151 @@ Requirements: Python >= 3.9.
+accepts a comma-separated list of formats to generate.
+The supported formats are:
+
+- ``html``: HTML documentation
+- ``pdf``: PDF documentation
+- ``epub``: ePUB documentation
+
+For example, to generate only HTML and PDF documentation, you can run:
+
+.. code-block:: shell
+
+    make build FORMATS=html,pdf
+
+To generate all supported formats, just omit the ``FORMATS`` argument:
+
+.. code-block:: shell
+
+    make build
+
+.. code-block:: shell
+
+    # This command will only generate HTML
+    make build FORMATS=html
+
+Building specific version
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, the ``make build`` command will generate documentation for
+all the versions defined in ``config.yml``.
+
+If you want to generate the documentation for a specific version,
+you can do so by using the ``VERSION`` argument. ``VERSION`` accepts any
+version that is specified in the ``config.yml`` file.
+
+For example, if you want to generate documentation for the
+``dev`` version, you can run:
+
+.. code-block:: shell
+
+    make build VERSION=dev
+
+This is useful if you only want to generate documentation for the
+version you are currently working on, or if you want to generate
+documentation for a specific version without having to rebuild all
+the other versions as well.
+
+Overriding a module of a version
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``make build`` command is programmed to generate the documentation
+for the modules that are defined in the ``config.yml`` file. Sometimes,
+it may be necessary to override the branch/remote of a module defined in the
+``config.yml`` file to build the documentation for a specific version
+or to test a specific commit/branch of a module.
+
+You can do so by using the ``MODULES`` argument. ``MODULES`` accepts a
+comma separated string where each item is of the following format:
+
+.. code-block:: text
+
+    version=<openwisp-version>,repository=<repo-owner>/<repo-name>,branch=<branch-name>
+
+E.g. if you want to build the documentation for the ``dev`` version, but want
+to use the ``feature`` branch of openwisp-controller of your fork, then
+the command will be:
+
+.. code-block:: shell
+
+    make build MODULES="version=dev:repository=<your-username>/openwisp-controller:branch=feature"
+
+The ``MODULES`` argument allows you to override the default settings
+for a single module, or multiple modules, defined in the
+``config.yml`` file.
+
+You can use the ``MODULES`` argument to add modules to a version that
+is not defined in the ``config.yml`` file.
+
+Building with SSH remotes
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, the OpenWISP modules are cloned over HTTPS. This may pose
+a hurdle if you wish to make changes to the cloned modules and push
+them to the remote URL. To use SSH remotes, you can set the environment
+variable ``DEV=1``. This will instruct the build to clone the modules
+using SSH instead of HTTPS. For example:
+
+.. code-block:: shell
+
+    DEV=1 make build
```

Comment:
Since we have a `dev` version, the choice of `DEV` here makes it confusing.
Why not just use `SSH=1`?